x-pack/plugins/alerting/server/types.ts

```diff
@@ -127,6 +127,32 @@ export interface RuleTypeParamsValidator<Params extends RuleTypeParams> {
   validateMutatedParams?: (mutatedOject: Params, origObject?: Params) => Params;
 }
 
+export interface GetSummarizedAlertsFnOpts {
+  start?: Date;
+  end?: Date;
+  executionUuid?: string;
+  ruleId: string;
+  spaceId: string;
+}
+
+// TODO - add type for these alerts when we determine which alerts-as-data
+// fields will be made available in https://github.com/elastic/kibana/issues/143741
+export interface SummarizedAlerts {
+  new: {
+    count: number;
+    alerts: unknown[];
+  };
+  ongoing: {
+    count: number;
+    alerts: unknown[];
+  };
+  recovered: {
+    count: number;
+    alerts: unknown[];
+  };
+}
+export type GetSummarizedAlertsFn = (opts: GetSummarizedAlertsFnOpts) => Promise<SummarizedAlerts>;
```

Comment:
I'm specifically requesting up to 1000 alerts in my query and setting `track_total_hits: true` to get the actual count. Since we allow querying by time range and we have an alert circuit breaker of 1000 alerts per run, if a rule that runs frequently and generates 1000 alerts each time asks for an alert summary, we want to limit the number of alert docs returned.

Reply:
gotcha, so `count != alerts.length` if > circuitbreaker alerts were generated.

Reply:
also if we're summarizing over the last day of alerts for a rule that runs every minute and (for example) generates 3 alerts each run. in that case we would over 4000 alerts in the last day and we'd get an exact count but only retrieve up to 1000.